pkg/runtime/reconciler.go

```diff
@@ -208,7 +208,6 @@ func (r *resourceReconciler) Sync(
 	var latest acktypes.AWSResource // the newly created or mutated resource
 
 	r.resetConditions(ctx, desired)
-	defer r.ensureConditions(ctx, latest, err)
```

Comment:
Would something like below behave differently?
```go
defer func() {
    r.ensureConditions(ctx, latest, err)
}()
```

Reply:
Yes. TIL...

Reply:
The irony here is that if we'd passed a *pointer* to an error and AWSResource interface, this would have worked :)